cmd/swarmctl/cluster/update.go

```diff
@@ -114,6 +119,11 @@ var (
 				if err != nil {
 					return err
 				}
+
+				if hbPeriod < time.Duration(0) {
```

Comment:
Where are you seeing this `uint64` conversion? In general, Duration should always be signed.